c++/src/Compression.cc

```diff
@@ -161,6 +133,8 @@ namespace orc {
       offset += currentSize;
       outputPosition += currentSize;
     }
+    BufferedOutputStream::BackUp(outputSize - outputPosition);
```

Comment:
Why should we do this?

Reply:
In the steam compressor, we will call writeData() multiple times, which requires the BufferedOutputStream to handle any remaining bytes correctly。